src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1360,6 +1457,88 @@ private Object writeReplace() {
             return new CollSer(CollSer.IMM_MAP, array);
         }
     }
+
+    static final class StableMap<K, V>
+            extends AbstractImmutableMap<K, V> {
```

Comment:
Same question about whether StableMap needs to go here. Though there’s more stuff going on for maps than lists here.